### PR TITLE
Adding audio context function for streaming buffers for sound emoji

### PR DESCRIPTION
**Pull request by riknoll**
Adding simulator support for streaming audio buffers for the micro:bit sound emoji.

Tested in Firefox, Safari, and Chrome on macOS

### CODE REVIEW

**Review by pelikhan — APPROVED** [review]

**Comment by abchatra** [comment]
@jaustin fyi

File: pxtsim/simlib.ts [hunk 1]
```diff
@@ -645,6 +645,74 @@ namespace pxsim {
             })
         }
 
+        const MAX_SCHEDULED_BUFFER_NODES = 3;
+
+        export function playPCMBufferStreamAsync(pull: () => Float32Array, sampleRate: number, volume = 0.3) {
```
Comment (abchatra) [inline pxtsim/simlib.ts:650]: Can you please add the reference to CODAL code this is trying to mirro?

**Review by abchatra — APPROVED** [review]

**Comment by pelikhan** [comment]
Should we also repipe this function to use your PCM api?

https://github.com/microsoft/pxt/blob/d0ec5d92f120492142330e264cff57ba5d5cb364/pxtsim/simlib.ts#L629

**Comment by riknoll** [comment]
@pelikhan no, I don't think it's necessary. Works well enough for the .wav files

**Pull request merged**